pkg/operator/status/status_controller.go

```diff
@@ -149,7 +158,16 @@ func (c StatusSyncer) Sync(ctx context.Context, syncCtx factory.SyncContext) err
 		return nil
 	}
 
-	clusterOperatorObj.Status.RelatedObjects = c.relatedObjects
+	if c.relatedObjectsFunc != nil {
+		ro := c.relatedObjectsFunc()
+		if ro != nil {
+			clusterOperatorObj.Status.RelatedObjects = ro
+		} else { // related objects unknown, carry over existing.
```

Comment:
i would still append the relatedObjects from constructor? it is not clear from the NewClusterOperatorStatusController() that when WithRelatedObjectsFunc() is used the relatedObjects passed to it are dropped.

Reply:
good point; updated.